### DISCUSSION TITLE
Add a `<Select.SelectAllTrigger>` component

### DISCUSSION THREAD

**Discussion by colinlienard** [chakra-ui/ark#2193] [Ideas]
For the Select, there is the `<Select.ClearTrigger>` component to clear the selection, but it would also be great to have a `<Select.SelectAllTrigger>` component (or something like that) to allow all options to be selected when the select is multiple.

It would be possible to do this with `api` but would be cleaner with a dedicated component in my opinion.

Also, I would love to try to implement it myself!

**Comment by pierresisson** [comment]
💯

**Comment by cschroeter** [comment]
@colinlienard 

Something like this? 
![image](https://github.com/chakra-ui/ark/assets/1846056/8a5b122b-2a45-4ade-9f0c-cd3c63b40fea)


or an dedicated button?

**Reply by colinlienard** [reply]
I think a dedicated button would do it, and then if people want to make it as an item within the dropdown they can style it and place it however they want @cschroeter

**Reply by pierresisson** [reply]
👀

**Reply by pierresisson** [reply]
@segunadebayo Any thoughts about this ? We would love to help! Wouldn't it be better to make a pull request directly?

**Reply by segunadebayo** [reply]
I think this should be implemented in userland code. Controlling the select component and explicitly selecting all item values when clicked.

I would consider putting together an example when I have time.

**Comment by pierresisson** [comment]
Hello @cschroeter, have you had a chance to look around? If not, can we help by contributing directly?

also, i would like thank you for your incredible work!

**Reply by segunadebayo** [reply]
I've just added support for this in [Zag.js](https://github.com/chakra-ui/zag/commit/761f185bc0e3fa337fcde6d3eeaae8e4a00ac00b) via `api.selectAll`

This will be supported in the next releases.
